### ISSUE TITLE
Messages Not Mapping To Receiver with `ClassNameRouter`

### ISSUE THREAD

**Issue by robskrob** [bernardphp/bernard#401]
I'm using bernard version `0.13.0`.

Whenever a publish [this message](https://github.com/robskrob/bernard-example/blob/feature/wordpress-sample/background-logic/consumer.php#L39) to the queue for consumption, I am seeing this error:

```
[bernard] caught exception Bernard\Exception\ReceiverNotFoundException::No receiver found for class "Bernard\Message\PlainMessage". while processing UpdateMacroCollection.
```

What's wrong with [how I'm mapping messages](https://github.com/robskrob/bernard-example/blob/feature/wordpress-sample/background-logic/message-consumer.php#L21) to their receivers? 

I was able to get everything working using `dev-master` but I did not like the idea of deploying code on a version like `dev-master` which is subject to change (seeing as I think it represents the latest?)

**Comment by robskrob** [comment]
OK I figured it out. [Where I was using](https://github.com/robskrob/bernard-example/blob/feature/wordpress-sample/background-logic/message-consumer.php#L21) `ClassNameRouter` I used `SimpleRouter` instead.

As a side, I am a little confused as to what an `Envelope` is. It looks like `ClassNameRouter` [has a dependency](https://github.com/bernardphp/bernard/blob/master/src/Router/ClassNameRouter.php#L5) on `Bernard\Envelope`. Moreover when I visit the docs the only place where I see anything about an `Envelope` mentioned is [here](https://bernard.readthedocs.io/producing.html#):
> It is also possible to get the queue directly from the queue factory and push the message there. But remember to wrap the message in an Envelope object.

Of course to figure out how to wire things together, I've been sifting through the source. I still cannot seem to grasp what the purpose and over all context of an `Envelope` is. Can someone expand on what an `Envelope` is, why it's used and how to use it?
